### PR TITLE
Fixes #8289 - cloudinit from rhev/ovirt support

### PR DESCRIPTION
**Pull request by karmab**
so this is the foreman part to handle cloud-init from rhev and as such user_data templates
it relies on the following pending PR
https://github.com/abenari/rbovirt/pull/48
and
https://github.com/fog/fog/pull/3250
i used the following user_data sample

``` json
#cloud-config
hostname: <%= @host.name %>
nicname: eth0
netmask: <%= @host.subnet.mask %>
gateway: <%= @host.subnet.gateway %> 
ip: <%= @host.ip %>
dns:
 - <%= @host.subnet.dns_primary %>
domain: <%= @host.domain %>
password: <%= root_pass %> 
ssh_authorized_keys:
 - ssh-rsa XXXX federer@bumblefoot
runcmd:
 - echo superbien > /tmp/prout1
 - echo 2280 > /tmp/prout2
```

### CODE REVIEW

**Comment by dLobatog** [comment]
[test]

**Comment by karmab** [comment]
the two dependant PR were merged, so this one can be evaluated now.
require rbovirt 0.0.31

File: app/models/compute_resource.rb [hunk 1]
```diff
@@ -134,7 +134,7 @@ def find_vm_by_uuid(uuid)
     client.servers.get(uuid) || raise(ActiveRecord::RecordNotFound)
   end
 
-  def start_vm(uuid)
+  def start_vm(uuid,user_data=nil)
```
Comment (ohadlevy) [inline app/models/compute_resource.rb:136]: why should we handle it differently from existing user data support

File: app/models/concerns/orchestration/compute.rb [hunk 2]
```diff
@@ -150,7 +150,12 @@ def delCompute
 
   def setComputePowerUp
     logger.info "Powering up Compute instance for #{name}"
-    compute_resource.start_vm uuid
+    if compute_resource.provider == 'Ovirt'
```
Comment (ohadlevy) [inline app/models/concerns/orchestration/compute.rb:153]: not sure i follow, this was not needed for others? cant we just eval if we have an image with user_data enabled and pass it when creating the vm in ovirt class?

Comment (karmab) [inline app/models/concerns/orchestration/compute.rb:153]: no for ovirt , cloudinit is passed upon start, not when creating the vm.

**Comment by unorthodoxgeek** [comment]
[test]

**Comment by karmab** [comment]
can you test this one, please :) ?

**Comment by domcleal** [comment]
Please update bundler.d/ovirt.rb to indicate the minimum rbovirt version, and once #2095 is merged then you'll also have the minimum fog required.

**Comment by karmab** [comment]
added the rbovirt dependency

**Comment by unorthodoxgeek** [comment]
@karmab - can you please rebase this with develop? there are merge conflicts.
robot, please [test]

**Comment by unorthodoxgeek** [comment]
[test]

**Comment by domcleal** [comment]
Tests can't happen if there's a merge conflict.

File: bundler.d/ovirt.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 group :ovirt do
-  gem 'rbovirt', '~> 0.0.24'
+  gem "rbovirt", ">= 0.0.31"
```
Comment (domcleal) [inline bundler.d/ovirt.rb:2]: Probably best to keep the `~>` as I think there are some incompatible changes planned for rbovirt.

**Comment by ReneCamu** [comment]
After applying this code snippets within foreman 1.7.2 the following error occurs:
Psych::SyntaxError
(<unknown>): did not find expected key while parsing a block mapping at line 16 column 2
app/models/compute_resources/foreman/model/ovirt.rb:144:in `start_with_cloudinit'
app/models/concerns/orchestration/compute.rb:155:in`setComputePowerUp'

Environment:
Foreman 1.7.2
ovirt: 0.0.31
fog: 1.27.4

Appreciate any help

**Comment by karmab** [comment]
let me rebase this once for all against latest develop and test again to fix
possible issues

**Comment by dLobatog** [comment]
[test]

**Comment by karmab** [comment]
guys closing this PR because too much dependencies too solve.
opening a new one right now againt latest develop ;)

**Comment by karmab** [comment]
@ReneCamu https://github.com/theforeman/foreman/pull/2198

**Pull request closed without merging**